front/app/components/SignUpIn/SignIn/PasswordSignin.tsx

```diff
@@ -159,53 +155,77 @@ class PasswordSignin extends PureComponent<
+    emailOrPhoneNumber: string | null,
+    password: string | null
+  ) {
     const {
       intl: { formatMessage },
-      tenant,
     } = this.props;
-    const { password } = this.state;
-    const phone =
-      !isNilOrError(tenant) && tenant.attributes.settings.password_login?.phone;
-    const hasEmailError = !phone && (!email || !isValidEmail(email));
-    const emailError = hasEmailError
-      ? formatMessage(messages.emailError)
-      : null;
-    const hasEmptyPasswordError = !password;
-
-    this.setState({ emailError, hasEmptyPasswordError });
-
-    if (emailError && this.emailInputElement) {
+    const hasValidPhoneNumber = emailOrPhoneNumber
+      ? isValidPhoneNumber(emailOrPhoneNumber)
+      : false;
+    const hasValidEmail = emailOrPhoneNumber
+      ? isValidEmail(emailOrPhoneNumber)
+      : false;
+
+    const hasEmailOrPhoneNumberValidationError = phoneLoginEnabled
+      ? !hasValidEmail && !hasValidPhoneNumber
+      : !hasValidEmail;
+    const hasEmptyPasswordError = password ? password.length === 0 : true;
+
+    this.setState({
+      emailOrPhoneNumberError: hasEmailOrPhoneNumberValidationError
+        ? formatMessage(
+            phoneLoginEnabled
+              ? messages.emailOrPhoneNumberError
+              : messages.emailError
+          )
+        : null,
+      hasEmptyPasswordError,
+    });
+
+    if (hasEmailOrPhoneNumberValidationError && this.emailInputElement) {
       this.emailInputElement.focus();
     }
 
-    if (!emailError && hasEmptyPasswordError && this.passwordInputElement) {
+    if (
+      !hasEmailOrPhoneNumberValidationError &&
+      hasEmptyPasswordError &&
+      this.passwordInputElement
+    ) {
       this.passwordInputElement.focus();
     }
 
-    return !emailError && !hasEmptyPasswordError;
+    return !hasEmailOrPhoneNumberValidationError && !hasEmptyPasswordError;
   }
 
-  handleOnSubmit = async (event: React.FormEvent) => {
-    event.preventDefault();
-
-    const { onSignInCompleted } = this.props;
-    const { formatMessage } = this.props.intl;
-    const { email, password } = this.state;
-
-    if (this.validate(email) && email && password) {
-      try {
-        this.setState({ processing: true });
-        const user = await signIn(email, password);
-        trackEventByName(tracks.signInEmailPasswordCompleted);
-        onSignInCompleted(user.data.id);
-      } catch (error) {
-        trackEventByName(tracks.signInEmailPasswordFailed, { error });
-        const signInError = formatMessage(messages.signInError);
-        this.setState({ signInError, processing: false });
+  handleOnSubmit =
+    (phoneLoginEnabled: boolean) => async (event: React.FormEvent) => {
+      event.preventDefault();
+
+      const { onSignInCompleted } = this.props;
+      const { formatMessage } = this.props.intl;
+      const { email, password } = this.state;
+
+      if (
+        this.validate(phoneLoginEnabled, email, password) &&
+        email &&
+        password
```

Comment:
Not ideal because after the validate call we know they're strings, but still better than using 'as string' in the signIn arguments (in case we change anything in the validate function).